probatus/binning/binning.py

```diff
@@ -437,9 +436,8 @@ def tree_bins(x, y, inf_edges, tree):
 
         bin_count = len(index)
 
-        tpf = TreePathFinder(tree)
-        boundaries = [bound["min"] for bound in tpf.get_boundaries().values()]
-        boundaries += [tpf.get_boundaries()[leaves[-1]]["max"]]
+        boundaries = np.unique(tree.tree_.threshold[tree.tree_.feature != _tree.TREE_UNDEFINED])
+        boundaries = [-np.inf] + boundaries.tolist() + [np.inf]
```

Comment:
Could you add a very simple unit test for that function:
- simple distribution consisting of 5 samples, where you can calculate where exactly the boundaries should be
- simple tree based model
- assert if the precalculated boundaries are compared to what the function returns?
- assert bin_count  parameter if they have the value correct as you precalculated

The current tests are a bit more general, and i doubt if they would catch a mistake in that line.

Reply:
add unit tests.